onnxscript/ir/passes/common/constant_manipulation.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Lift constants to initializers."""
+
+from __future__ import annotations
+
+__all__ = [
+    "LiftConstantsToInitializersPass",
+]
+
+import logging
+
+import numpy as np
+
+from onnxscript import ir
+
+logger = logging.getLogger(__name__)
+
+
+class LiftConstantsToInitializersPass(ir.passes.InPlacePass):
+    def call(self, model: ir.Model) -> ir.passes.PassResult:
+        """Convert constant nodes from node belonged graph to its initializers."""
+        count = 0
+        for node in ir.traversal.RecursiveGraphIterator(model.graph):
+            if node.op_type != "Constant" or node.domain not in ("", "onnx.ai"):
+                continue
+
+            constant_node_attribute = set(node.attributes.keys())
+            if len(constant_node_attribute) != 1:
+                logger.debug(
+                    "Invalid constant node '%s' has more than one attribute", node.name
+                )
+                continue
+
+            attr_name, attr_value = next(iter(node.attributes.items()))
+            initializer_name = node.outputs[0].name
+            assert initializer_name is not None
+            assert isinstance(attr_value, ir.Attr)
+            tensor = _constant_node_attribute_to_tensor(
+                attr_name, attr_value, initializer_name
+            )
+            if tensor is None:
+                logger.debug(
+                    "Invalid constant node '%s' has unsupported attribute value", node.name
+                )
+                continue
+            # Register an initializer with the tensor value
+            initializer = ir.Value(
+                name=initializer_name,
+                shape=tensor.shape,  # type: ignore[arg-type]
+                type=ir.TensorType(tensor.dtype),
+                const_value=tensor,
+            )
+            assert node.graph is not None
+            assert isinstance(node.graph, ir.Graph)
```

Comment:
@justinchuby does this make sense? I think there should not be any ir.Function node coming out from recursive iterator?

Reply:
Does make sense. Thanks!

Reply:
Technically the graph in a function is private and not used directly. It is currently an implementation detail

Reply:
sorry I was wrong. It is pointed to a graph when we call `function.append`, but it is not when we call ir.Node(graph=function). I need to figure out how to reconcile this. Suggestions appreciated. https://github.com/microsoft/onnxscript/issues/2181

Reply:
https://github.com/microsoft/onnxscript/pull/2183